### DISCUSSION TITLE
BMS minerals no communication esp

### DISCUSSION THREAD

**Discussion by franto262** [syssi/esphome-seplos-bms#103] [Q&A] [answered]
Hello
I will ask for help

My bms did not send data to esp.
converter rs485 txd led blink

https://www.aliexpress.com/item/1005005615276887.html?spm=a2g0o.order_list.order_list_main.167.3ad3180293cBax

![1718467654069](https://github.com/syssi/esphome-seplos-bms/assets/135267638/6599ff85-16f4-498b-ae1d-07903d378bd5)
![IMG_20240616_132212](https://github.com/syssi/esphome-seplos-bms/assets/135267638/ea6b64db-78c4-4c8b-b0b5-5ad318667266)

communication from the PC works
![Bez názvu](https://github.com/syssi/esphome-seplos-bms/assets/135267638/b7ded224-7357-4e6a-9254-306f3419e64c)

my YAML

`substitutions:
  name: seplos-bms
  device_description: "Monitor a Boqiang BMS via RS485"
  external_components_source: github://syssi/esphome-seplos-bms@main
  tx_pin: GPIO4
  rx_pin: GPIO5

esphome:
  name: baterkabms-280
  friendly_name: BaterkaBMS 280

esp8266:
  board: esp01_1m

external_components:
  - source: ${external_components_source}
    refresh: 0s

logger:
  level: DEBUG

uart:
  id: uart_0

  baud_rate: 19200
  tx_pin: ${tx_pin}
  rx_pin: ${rx_pin}
 
  rx_buffer_size: 384
  debug:
    dummy_receiver: false
    direction: BOTH
    after:
      delimiter: "\r"
    sequence:
      - lambda: UARTDebug::log_string(direction, bytes);

seplos_modbus:
  id: modbus0
  uart_id: uart_0
  rx_timeout: 150ms

seplos_bms:
  id: bms0
 
 

  address: 0x00
  protocol_version: 0x20
  override_pack: 1
  override_cell_count: 10
  seplos_modbus_id: modbus0
  update_interval: 10s

sensor:
  - platform: seplos_bms
    min_cell_voltage:
      name: "${name} min cell voltage"
    max_cell_voltage:
      name: "${name} max cell voltage"
    min_voltage_cell:
      name: "${name} min voltage cell"
    max_voltage_cell:
      name: "${name} max voltage cell"
    delta_cell_voltage:
      name: "${name} delta cell voltage"
    average_cell_voltage:
      name: "${name} average cell voltage"
    cell_voltage_1:
      name: "${name} cell voltage 1"
    cell_voltage_2:
      name: "${name} cell voltage 2"
    cell_voltage_3:
      name: "${name} cell voltage 3"
    cell_voltage_4:
      name: "${name} cell voltage 4"
    cell_voltage_5:
      name: "${name} cell voltage 5"
    cell_voltage_6:
      name: "${name} cell voltage 6"
    cell_voltage_7:
      name: "${name} cell voltage 7"
    cell_voltage_8:
      name: "${name} cell voltage 8"
    cell_voltage_9:
      name: "${name} cell voltage 9"
    cell_voltage_10:
      name: "${name} cell voltage 10"
    cell_voltage_11:
      name: "${name} cell voltage 11"
    cell_voltage_12:
      name: "${name} cell voltage 12"
    cell_voltage_13:
      name: "${name} cell voltage 13"
    cell_voltage_14:
      name: "${name} cell voltage 14"
    cell_voltage_15:
      name: "${name} cell voltage 15"
    cell_voltage_16:
      name: "${name} cell voltage 16"
    temperature_1:
      name: "${name} temperature 1"
    temperature_2:
      name: "${name} temperature 2"
    temperature_3:
      name: "${name} temperature 3"
    temperature_4:
      name: "${name} temperature 4"
    temperature_5:
      name: "${name} environment temperature"
    temperature_6:
      name: "${name} mosfet temperature"
    total_voltage:
      name: "${name} total voltage"
    current:
      name: "${name} current"
    power:
      name: "${name} power"
    charging_power:
      name: "${name} charging power"
    discharging_power:
      name: "${name} discharging power"
    residual_capacity:
      name: "${name} residual capacity"
    battery_capacity:
      name: "${name} battery capacity"
    rated_capacity:
      name: "${name} rated capacity"
    state_of_charge:
      name: "${name} state of charge"
    charging_cycles:
      name: "${name} charging cycles"
    state_of_health:
      name: "${name} state of health"
    port_voltage:
      name: "${name} port voltage"`

[logs_baterkabms-280_run.txt](https://github.com/user-attachments/files/15857646/logs_baterkabms-280_run.txt)

**Comment by franto262** [comment]
when I connect at the same time pc also
data is flowing

[logs_baterkabms-280_run (3).txt](https://github.com/user-attachments/files/15857982/logs_baterkabms-280_run.3.txt)

**Reply by syssi** [reply, marked as answer]
The PC polls the BMS periodically. Please compare the requests of the PC with the ones of the ESP. Is there a difference? If not your RS485 converter is probably broken and unable to pass the requests to the BMS.

**Reply by franto262** [reply]
My rs485 converter replace and it already works .I did not think that the converter would be damaged,was new.

Thanks for help
